lib/Collaboration/Resources/ResourceProviderCard.php

```diff
@@ -0,0 +1,134 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Deck\Collaboration\Resources;
+
+
+use OCA\Deck\Db\Acl;
+use OCA\Deck\Db\Board;
+use OCA\Deck\Db\BoardMapper;
+use OCA\Deck\Service\PermissionService;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\AppFramework\QueryException;
+use OCP\Collaboration\Resources\IManager;
+use OCP\Collaboration\Resources\IProvider;
+use OCP\Collaboration\Resources\IResource;
+use OCP\Collaboration\Resources\ResourceException;
+use OCP\IUser;
+
+class ResourceProviderCard implements IProvider {
+
+	const RESOURCE_TYPE = 'deck-card';
+
+	private $boardMapper;
+	private $permissionService;
+
+	/** @var array */
+	protected $nodes = [];
+
+	public function __construct(BoardMapper $boardMapper, PermissionService $permissionService) {
+		$this->boardMapper = $boardMapper;
+		$this->permissionService = $permissionService;
+	}
+
+	/**
+	 * Get the type of a resource
+	 *
+	 * @param IResource $resource
+	 * @return string
+	 * @since 15.0.0
+	 */
+	public function getType(): string {
+		return self::RESOURCE_TYPE;
+	}
+
+	/**
+	 * Get the rich object data of a resource
+	 *
+	 * @param IResource $resource
+	 * @return array
+	 * @throws \OCP\AppFramework\Db\DoesNotExistException
+	 * @throws \OCP\AppFramework\Db\MultipleObjectsReturnedException
+	 * @since 16.0.0
+	 */
+	public function getResourceRichObject(IResource $resource): array {
+		$board = $this->getBoard($resource);
+		//$link = \OC::$server->getURLGenerator()->linkToRoute('deck.page.index') . '#/board/' . $resource->getId();
+		$link = \OC::$server->getURLGenerator()->linkToRoute('deck.page.index') ;
+
+		return [
+			'type' => self::RESOURCE_TYPE,
+			'id' => $resource->getId(),
+			'name' => $board->getTitle(),
+			'link' => $link,
+			'iconUrl' => \OC::$server->getURLGenerator()->imagePath('deck', 'deck-dark.svg')
+		];
+
+	}
+
+	/**
+	 * Can a user/guest access the collection
+	 *
+	 * @param IResource $resource
+	 * @param IUser|null $user
+	 * @return bool
+	 * @since 16.0.0
+	 */
+	public function canAccessResource(IResource $resource, ?IUser $user): bool {
+		if ($resource->getType() !== self::RESOURCE_TYPE || !$user instanceof IUser) {
+			return false;
+		}
+		$board = $this->getBoard($resource);
```

Comment:
The issue here is that you try to fetch the board but you have the card id. So the board is not found and the canAccessResource returns false, so the resource is never added to the project.

Reply:
Thanks for your review. 
The problem is, I can choose a card, but it would not shown
![grafik](https://user-images.githubusercontent.com/24530680/67084575-81b82800-f19d-11e9-9a7b-4f37c55d1f58.png)
show project details:
![grafik](https://user-images.githubusercontent.com/24530680/67084640-a3191400-f19d-11e9-9f60-44a52a9ed0bb.png)

And if I change the canAccessResource function to "return true;" there is no change

Reply:
So it is not just the canAccessResource method but also getResourceRichObject

You need to adjust the implementation to properly fetch the card details. Right now when linking to a card, you add the card id but try to fetch details for the board with that id. It cannot work that way. Please always checkout the nextcloud log in such cases as the error message gives a clear hint on why the request fails:

![image](https://user-images.githubusercontent.com/3404133/67165103-5aba4b80-f381-11e9-886a-ff34280dd9cf.png)

```
"Exception":"Error","Message":"Call to a member function getTitle() on null","Code":0,"Trace":[{"file":"/var/www/html/lib/private/Collaboration/Resources/Manager.php","line":302,"function":"getResourceRichObject","class":"OCA\\Deck\\Collaboration\\Resources\\ResourceProviderCard"
```